packages/contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -371,6 +371,9 @@ contract OptimismPortal is Initializable, ResourceMetering, Semver {
             );
         }
 
+        // Prevent depositing transactions that have too small of a gas limit.
+        require(_gasLimit >= 21_000, "OptimismPortal: gas limit must cover instrinsic gas cost");
```

Comment:
I think this should really be `21_000 + _data.length` or similar

Reply:
I see your point here. I meant to just set a lower bound. It would be more accurate to count the zero and non zero bytes and multiply them by their gas prices and sum the products together. Not sure if all that iteration is worth it, so just adding the data length is a heuristic that underestimates. We could multiply by 16 (way overestimate) or 12 (prob slight overestimate)

Reply:
I think that the L1 on chain burn + L1 data cost is sufficient dos protection. It buys simplicity for automation, ie contracts that do deposits dont need to have dynamic gas limit setting